metals/src/main/scala/scala/meta/internal/metals/codeactions/ImportMissingSymbol.scala

```diff
@@ -97,9 +113,12 @@ class ImportMissingSymbol(compilers: Compilers) extends CodeAction {
             case diag @ ScalacDiagnostic.SymbolNotFound(name)
                 if params.getRange().overlapsWith(diag.getRange()) =>
               importMissingSymbol(diag, name)
+            // `foo.xxx` where `xxx` is not a member of `foo`
+            // we search for `xxx` only when the target is Scala3
+            // considering there might be an extension method.
             case d @ ScalacDiagnostic.NotAMember(name)
-                if params.getRange().overlapsWith(d.getRange()) =>
-              importMissingSymbol(d, name)
+                if isScala3 && params.getRange().overlapsWith(d.getRange()) =>
```

Comment:
This change is a bit off-topic from the PR title. Now, `"xxx" is not a member of ...` won't trigger the code action for nothing in Scala2 :+1: